File: data/code_pairs/fortran/46.f90
! Module definition
module assignment_module
    implicit none
    real :: XYTRAN_TRCOEFFS = 0.0
    real :: XYFIT_TRCOEFFS = 5.0
contains
    ! The subroutine to perform the assignment
    subroutine perform_assignment()
        XYTRAN_TRCOEFFS = XYFIT_TRCOEFFS
    end subroutine perform_assignment
end module assignment_module

! Test program
program test_program
    use assignment_module
    implicit none
    call perform_assignment()
    if (XYTRAN_TRCOEFFS == XYFIT_TRCOEFFS) then
        print *, "Test passed."
    else
        print *, "Test failed."
    end if
end program test_program

File: data/code_pairs/cpp/46.cpp
#include <iostream>
#include <cmath>

// Global variables
double XYTRAN_TRCOEFFS = 0.0;
double XYFIT_TRCOEFFS = 5.0;

// Function to perform the assignment
void perform_assignment() {
    XYTRAN_TRCOEFFS = XYFIT_TRCOEFFS;
}

// Main function for testing
int main() {
    // Run the assignment function
    perform_assignment();

    // Testing function
    if (std::fabs(XYTRAN_TRCOEFFS - XYFIT_TRCOEFFS) < 1e-6) {
        std::cout << "Test passed." << std::endl;
    } else {
        std::cout << "Test failed." << std::endl;
    }

    return 0;
}